### DISCUSSION TITLE
Renaming variables

### DISCUSSION THREAD

**Discussion by mauronz** [Vector35/binaryninja-api#1757] [General] [answered]
Is there a way to permanently rename function variables? From what I'm seeing we only have access to read-only Variable objects

**Comment by rssor** [comment, marked as answer]
Renaming is done by simply redefining the variable. The wrapper objects are read-only, but you can pass them back in through `Function.create_user_var` to replace what was already there.

**Comment by mauronz** [comment]
Works perfectly, thank you
